src/components/Firestore/DocumentPreview/index.tsx

```diff
@@ -67,14 +67,18 @@ const DocumentPreview: React.FC<Props> = ({
                 setIsAddingField(false);
               }}
               onSave={(key, value) => {
-                updateField(reference, data as FirestoreMap, [key], value);
+                if (!data) {
+                  reference.set({ [key]: value });
```

Comment:
Please moving this into `./api.ts`. Also worth considering adding a precondition to assert that the document actually doesn't exist to prevent accidentally overwriting data, but that can be in a follow up PR.